### DISCUSSION TITLE
Error on upgrading Airflow from 1.10.11 to Airflow 1.10.15

### DISCUSSION THREAD

**Discussion by mavenzer** [apache/airflow#25234] [Q&A]
Hi, 

I'm trying to upgrade airflow 1.10.11 to Airflow 1.10.15(a.k.a)  bridge-release as I don't want to disrupt the current workflow.
But while upgrading airflow to 1.10.15 I'm getting the following errors :

Deployment Platform: OpenShift v3
Python Version: 3.8
OS:  RHEL 

```
ERROR: The `secret_key` setting under the webserver config has an insecure value - Airflow has 
failed safe and refuses to start. Please change this value to a new, per-environment,
 randomly generated string, for example using this command `openssl rand -hex 30`
``` 

So according to my understanding, I'm using the static fernet key in my deployment, which should be generated randomly.
My current airflow fernet key is as follows which is being used by Airflow 1.10.11
```
apiVersion:v1
kind:Secret
metadata:
    name : airflow-secret
    namespace : CUSTOM_NAMESPACE
    labels: 
        app:airflow
type: Opaque
stringData:
   fernet-key: my_fernet_key

```

My deployment config for Airflow is as follows : 
```
apiVersion: v1
kind: DeploymentConfig
metadata:
  name: airflow-webserver
  namespace: CUSTOM_NAMESPACE
  labels:
    app: airflow
spec:
  strategy: 
    type: Rolling
  trigger: 
    - type : ConfigChange
    - type : ImageChange
      ImageChangeParams: 
        automatic: true
        containerNames:
        - airflow-webserver
        from: 
          kind: ImageStreamTag
          namespace: CUSTOM_NAMESPACE
  replicas: 1
  revisionHistoryLimit : 10
  paused: false
  selector :
    app : airflow
    deploymentconfig : airflow-webserver
  template:
    metadata:
      labels:
        name: airflow-webserver
        app: airflow
        deploymentconfig : airflow-webserver
    spec: 
      volumes: 
      - name: airflow-dags
      persistentVolumeClaims:
        claimName: airflow-dags
    containers:
    - name:  airflow-webserver
      image:  airflow:latest
      resources:  
        limits:
          memory: 4Gi
      env:
        - name : FERNET_KEY
          valueFrom:
            secretKeyRef: 
              name: airflow-secrets
              key : fernet-key
        - name : SERVICE_ACCOUNT_NAME
          valueFrom:
            secretKeyRef:
              name: airflow-service-account
              key : service-account-name
      ports:
        - containerPort: 8080
          protocol: TCP

      volumeMounts:
        - name: airflow-dags
          mountPath: /opt/airflow/dags
        - name: airflow-logs
          mountPath: /opt/airflow/logs
```

My general understanding is I need to provide a randomly generated fernet key in the deployment config of fernet key. Any possible way to resolve the error.

**Comment by potiuk** [comment]
You can provide ANY random key. As long as it is not "my_fernet_key". Just change it.. Howver you should follow instruction on providing old and new fernet key at the same time if you do not want to loose old data (please read the docs of airflow that explain that in detail.

**Comment by potiuk** [comment]
Reading docs about fernet key is generally a good start.

**Comment by mavenzer** [comment]
So I generated the fernet key using the following command

```
from cryptography. fernet import Fernet
fernet_key = Fernet. generate_key()
print(fernet_key)
```

And put this fernet key in the YAML file of Airflow secrets as mentioned in the above post.

But still getting the same error  as above while deploying the webserver 
```
ERROR: The `secret_key` setting under the webserver config has an insecure value - Airflow has 
failed safe and refuses to start. Please change this value to a new, per-environment,
 randomly generated string, for example using this command `openssl rand -hex 30`
```
My airflow config is as follows i.e in **airflow.cfg**

```
fernet_key = $FERNET_KEY
```

**Reply by potiuk** [reply]
No idea about your setup but you are likely doing something wrong - for example your FERNET_KEY variable is not set in the environment of your airflow environment. I'd encourage you to thorougly read and understand https://airflow.apache.org/docs/apache-airflow/stable/howto/set-config.html and try out various way of settings different variables (that might also help you to understand how to set them and update them in Airflow 2 because it has not changed much in Airflow 2. You just need to experiment more and get to understand which config values you have. 

The change in question was implemented here: https://github.com/apache/airflow/pull/12747/files and it only displays the warning if your fernet key has the expicit value "temporary_key" . This indicates that you simply misunderstand some parts of airflow configuration and your configuration is not properly applied. But this also indicates that you have likely bigger problems and making sure how airflow configuration works and how to apply any changes so that Airlfow components get it si an absolute prerequisite for you to not only migrate to new Airlfow version but also to mange it in the future.

You really need to debug and understand yourself, this is the best way you can learn it (and you will need it).

**Comment by mavenzer** [comment]
The issue can be resolved by generating new **secret_key** in the **airflow.cfg** file.

The process of generating the secret_key is given in the error message, by using the following command
```
openssl rand -hex 30 
```
and put the corresponding value in the 
```
secret_key  =  Value generated from the above command 
```
